AltStore/Settings/LicensesViewController.swift

```diff
@@ -8,46 +8,90 @@
 
 import UIKit
 
-class LicensesViewController: UIViewController
+class LicensesViewController: UITableViewController
 {
-    private var _didAppear = false
-    
-    @IBOutlet private var textView: UITextView!
+    private var licenses: [LicenseItem] = []
     
     override var preferredStatusBarStyle: UIStatusBarStyle {
         return .lightContent
     }
-    
-    override func viewWillAppear(_ animated: Bool)
+
+    override func viewDidLoad()
     {
-        super.viewWillAppear(animated)
-        
-        self.view.setNeedsLayout()
-        self.view.layoutIfNeeded()
-        
-        // Fix incorrect initial offset on iPhone SE.
-        self.textView.contentOffset.y = 0
+        super.viewDidLoad()
+        loadLicenses()
     }
-    
-    override func viewDidAppear(_ animated: Bool)
+
+    private func loadLicenses()
+    {
+        guard let path = Bundle.main.path(forResource: "licenses", ofType: "json") else {
+            dismiss(animated: true)
+            return
+        }
+
+        let url = URL(fileURLWithPath: path)
+
+        guard let data = try? Data(contentsOf: url), let json = try? JSONDecoder().decode([LicenseItem].self, from: data) else {
+            dismiss(animated: true)
+            return
+        }
+
+        licenses = json
+    }
+}
+
+extension LicensesViewController
+{
+    override func numberOfSections(in tableView: UITableView) -> Int
     {
-        super.viewDidAppear(animated)
-        
-        _didAppear = true
+        return 1
     }
 
-    override func viewDidLayoutSubviews()
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
     {
-        super.viewDidLayoutSubviews()
-        
-        self.textView.textContainerInset.left = self.view.layoutMargins.left
-        self.textView.textContainerInset.right = self.view.layoutMargins.right
-        self.textView.textContainer.lineFragmentPadding = 0
-        
-        if !_didAppear
+        return licenses.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
+    {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "licenseListCell", for: indexPath) as! LicenseTableViewCell
+        let license = licenses[indexPath.row]
+
+        setStyle(for: cell, at: indexPath)
+
+        cell.productLabel.text = license.product
+        cell.authorLabel.text = license.author
+
+        return cell
+    }
+
+    private func setStyle(for cell: LicenseTableViewCell, at indexPath: IndexPath)
+    {
+        switch indexPath.row {
+        case 0:
+            cell.style = .top
+            break
+        case licenses.count - 1:
+            cell.style = .bottom
+            break
```

Comment:
We don't need to declare `break`. Please remove it.

Reply:
Thank you for the review. You're right hehe.

Right now the PR can't be merged and since it has been opened for months without answer from Riley I'm not in a rush to update it. I will get to updating it eventually though and I will take your recommendation into account 😄